### PR TITLE
// TODO: Print out all the sub-commands available

### PR DESCRIPTION
**Pull request by bradyjoestar**
* [ ] Updated all relevant documentation in docs
* [ ] Updated all code comments where relevant
* [ ] Wrote tests
* [ ] Updated CHANGELOG_PENDING.md

### CODE REVIEW

**Comment by melekes** [comment]
Hi! Thank you for your contribution! 

Is there an issue? Could you expand on what this PR does?

**Comment by bradyjoestar** [comment]
Sure.You are welcome,it's my pleasure.

no,there isn't a issue.
I searched all the "TODO" in the code and found it in "abci-cli.go"

The following picture is we use `abci-cli console` and then executed `info` and `nana`(unimplement command):



I think the `TODO: Print out all the sub-commands available` want to print the all available command in the console.
```
func cmdUnimplemented(cmd *cobra.Command, args []string) error {
	// TODO: Print out all the sub-commands available
	msg := "unimplemented command"
	if err := cmd.Help(); err != nil {
		msg = err.Error()
	}
	if len(args) > 0 {
		msg += fmt.Sprintf(" args: [%s]", strings.Join(args, " "))
	}
	printResponse(cmd, args, response{
		Code: codeBad,
		Log:  msg,
	})
	return nil
}
```

so I changed it, after changed:

File: abci/cmd/abci-cli/abci-cli.go [hunk 1]
```diff
@@ -477,9 +503,8 @@ func muxOnCommands(cmd *cobra.Command, pArgs []string) error {
 }
 
 func cmdUnimplemented(cmd *cobra.Command, args []string) error {
-	// TODO: Print out all the sub-commands available
 	msg := "unimplemented command"
-	if err := cmd.Help(); err != nil {
+	if err := muxonCmd.Help(); err != nil {
```
Comment (melekes) [inline abci/cmd/abci-cli/abci-cli.go]: I think we can do it without any additional commands. What if we just cycle through available commands and print their use and short description

```
fmt.Println("Available commands:")
fmt.Printf("%s: %s\n", echoCmd.Use, echoCmd.Short)
...
```

Comment (bradyjoestar) [reply abci/cmd/abci-cli/abci-cli.go]: I had made it more complicated.  lol

Thank you!
By the way，I'm reading the code of consensus which is really awesome!!

File: abci/cmd/abci-cli/abci-cli.go [hunk 2]
```diff
@@ -477,11 +477,17 @@ func muxOnCommands(cmd *cobra.Command, pArgs []string) error {
 }
 
 func cmdUnimplemented(cmd *cobra.Command, args []string) error {
-	// TODO: Print out all the sub-commands available
 	msg := "unimplemented command"
-	if err := cmd.Help(); err != nil {
-		msg = err.Error()
-	}
+
+	fmt.Println("Available commands:")
```
Comment (melekes) [inline abci/cmd/abci-cli/abci-cli.go]: let's move this section after `printResponse`

File: abci/cmd/abci-cli/abci-cli.go [hunk 3]
```diff
@@ -477,11 +477,17 @@ func muxOnCommands(cmd *cobra.Command, pArgs []string) error {
...
+	fmt.Printf("%s: %s\n", echoCmd.Use, echoCmd.Short)
+	fmt.Printf("%s: %s\n", infoCmd.Use, infoCmd.Short)
+	fmt.Printf("%s: %s\n", checkTxCmd.Use, checkTxCmd.Short)
+	fmt.Printf("%s: %s\n", deliverTxCmd.Use, deliverTxCmd.Short)
+	fmt.Printf("%s: %s\n", queryCmd.Use, queryCmd.Short)
+	fmt.Printf("%s: %s\n", commitCmd.Use, commitCmd.Short)
+	fmt.Printf("%s: %s\n", setOptionCmd.Use, setOptionCmd.Short)
```
Comment (melekes) [inline abci/cmd/abci-cli/abci-cli.go]: Also, would be good to print at the end

```
fmt.Println("Use \"[command] --help\" for more information about a command.")
```

Comment (bradyjoestar) [reply abci/cmd/abci-cli/abci-cli.go]: Yes sir!

**Pull request merged**